include/ogdf/basic/internal/graph_iterators.h

```diff
@@ -40,61 +40,67 @@ namespace ogdf {
 
 namespace internal {
 
-template<class GraphObjectPtr, bool isReverse>
+template<class GraphObjectPtr, bool isReverse, bool isConst>
 class GraphIteratorBase;
 template<class GraphObjectPtr>
-using GraphIterator = GraphIteratorBase<GraphObjectPtr, false>;
+using GraphIterator = GraphIteratorBase<GraphObjectPtr, false, false>;
 template<class GraphObjectPtr>
-using GraphReverseIterator = GraphIteratorBase<GraphObjectPtr, true>;
+using GraphConstIterator = GraphIteratorBase<GraphObjectPtr, false, true>;
+template<class GraphObjectPtr>
+using GraphReverseIterator = GraphIteratorBase<GraphObjectPtr, true, false>;
```

Comment:
I tried introducing the whole const iterator concept to graph lists while making them behave like standard iterators following the same way of thinking. 😅  This then broke the non-const method `const GraphObjectPtr& GraphIteratorBase::operator*()` for some weird reason I no longer recall (see the change disabling it [here](https://github.com/ogdf/ogdf/commit/c38e38c9422625bd378c402ce6c2fdb5718bf8b8#diff-72053da2e3249e43a9905b7d3c742e8dd74c4991b16206da838616bc4ba6b932), unfortunately I no longer find the failing CI run before that) - probably the method was already broken as it was, the change to const iterators just surfaced this by breaking compilation. Constness probably makes little sense anyways with GraphList iterators, because neither the list nor the contained objects allows public modification. I guess this whole tangle of classes used for graph element storage (esp. the empty `GraphObjectContainer` class or the list destructors!) needs a further clean-up, but that should be done in another PR. The only usage of these `using`s is in GraphList.h anyways (and the std-lib now introduced its own wrappers for reversing and const iterators), so they could probably also go as a whole...